pkg/chunked/dump/dump.go

```diff
@@ -0,0 +1,219 @@
+package dump
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/containers/storage/pkg/chunked/internal"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	NOESCAPE_SPACE = 1 << iota
+	ESCAPE_EQUAL
+	ESCAPE_LONE_DASH
+	ESCAPE_STANDARD
+)
+
+func escaped(val string, length int, escape int) string {
+	noescapeSpace := escape&NOESCAPE_SPACE != 0
+	escapeEqual := escape&ESCAPE_EQUAL != 0
+	escapeLoneDash := escape&ESCAPE_LONE_DASH != 0
+
+	if length < 0 {
+		length = len(val)
+	}
+
+	if escapeLoneDash && length == 1 && val[0] == '-' {
```

Comment:
In golang, this is nicer as `escapeLoneDash && val == "-"`, at least once you drop the length arg.